matrixbot/plugins/wkbotsfeeder.py

```diff
@@ -128,3 +136,41 @@ def command(self, sender, room_id, body, handler):
     def help(self, sender, room_id, handler):
         self.logger.debug("WKBotsFeederPlugin help")
         return
+
+def selftest():
+    print("selftest: " + os.path.basename(__file__))
+    settings = {
+        "name": "wk",
+        "last_buildjob_url_schema": "https://build.webkit.org/#/builders/%d/builds/%d",
+        "builds_url_schema": "https://build.webkit.org/api/v2/builders/%d/builds?complete=true&order=-number&limit=1",
+        "only_failures": False,
+        "rooms": ["0"],
+        "builders": {
+            "GTK-Linux-64-bit-Release-Ubuntu-LTS-Build": {
+                "builderid": 68,
+            },
+        },
+    }
+    plugin = WKBotsFeederPlugin(utils.MockBot(), settings)
+
+    test_async(plugin)
+    test_can_fetch_last_build(plugin)
+
+def test_async(plugin):
+    print("test_async: ")
+    logging.basicConfig(level = logging.DEBUG)
+    plugin.lasttime = 0
+    plugin.period = 0
+    plugin.async()
+    print("")
+    print("Ok")
+
+def test_can_fetch_last_build(plugin):
+    put("test_can_fetch_last_build: ")
```

Comment:
What is the idea behind  use `sys.stdout.write` instead of a simple `print()`

Reply:
`print()` always prints a newline character. `sys.stdout.write()` prints without a new character.